app/main/actions/search.js

```diff
@@ -36,15 +36,24 @@ const DEFAULT_SCOPE = {
  * @param {Function} callback Callback function that receives used search term and found results
  */
 const eachPlugin = (term, display) => {
+  const externalSettings = config.get('external') || {}
   // TODO: order results by frequency?
   Object.keys(plugins).forEach(name => {
+    const settings = externalSettings[name]
+    if (settings) {
+      Object.keys(settings).forEach(key => {
+        settings[key] = settings[key].value
```

Comment:
Yes that's better, but for each setting we still need to keep a `type`, `description`, `defaultValue` and other specific settings like `multi` and `options`. So the actual structure is closer to:
```javascript
{
  pluginName: {
    language: {
      type: 'option',
      options: ['en', 'fr'],
      value: 'en'
    },
    token: {
     type: 'string',
      value: 'user-token'
    }
  }
}
```